### ISSUE TITLE
Plugin fails to load: uninitialized constant TestCenter::Helper::MultiScanManager::TestBatchWorker

### ISSUE THREAD

**Issue by dionc** [lyndsey-ferguson/fastlane-plugin-test_center#170]
### New Issue Checklist

- [x] Updated `fastlane-plugin-test_center` to the latest version
- [x] I read the [README.md](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md)
- [x] I reviewed the [example(s)](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/blob/master/README.md) for the action(s) I am using
- [x] I have removed any sensitive data such as passwords, authentication tokens, or anything else I do not want to world to see

### Issue Description

With the latest version of fastlane (2.135.2) and the latest version of fastlane-plugin-test_center (3.8.13), the plugin fails to load with the following error:
```Error loading plugin 'fastlane-plugin-test_center': uninitialized constant TestCenter::Helper::MultiScanManager::TestBatchWorker```

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>> bundle exec fastlane update_plugins --verbose --capture-output
[✔] 🚀 
DEBUG [2019-11-15 09:58:57.33]: Checking if there are any plugins that should be loaded...
DEBUG [2019-11-15 09:58:57.33]: Loading 'fastlane-plugin-test_center' plugin
INFO [2019-11-15 09:58:57.33]: gem 'fastlane-plugin-test_center' is already installed
ERROR [2019-11-15 09:58:57.36]: Error loading plugin 'fastlane-plugin-test_center': uninitialized constant TestCenter::Helper::MultiScanManager::TestBatchWorker
ERROR [2019-11-15 09:58:57.36]: It seems like you wanted to load some plugins, however they couldn't be loaded
ERROR [2019-11-15 09:58:57.36]: Please follow the troubleshooting guide: https://docs.fastlane.tools/plugins/plugins-troubleshooting/
+-----------------------------+---------+------------------+
|                       Used plugins                       |
+-----------------------------+---------+------------------+
| Plugin                      | Version | Action           |
+-----------------------------+---------+------------------+
| fastlane-plugin-test_center | 3.8.13  | No actions found |
+-----------------------------+---------+------------------+

[!] No actions were found while loading one or more plugins
    Please use `bundle exec fastlane` with plugins
    More info - https://docs.fastlane.tools/plugins/using-plugins/#run-with-plugins
</pre>
</details>

### Environment

<details>
  <pre><details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.15                                                          |
| Ruby                        | 2.6.3                                                          |
| Bundler?                    | true                                                           |
| Git                         | git version 2.21.0 (Apple Git-122.2)                           |
| Installation Source         | /Volumes/90Seconds/ios/vendor/bundle/ruby/2.6.0/bin/fastlane   |
| Host                        | Mac OS X 10.15 (19A578c)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib |
| OpenSSL Version             | LibreSSL 2.8.3                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version               | 11.2.1                                                         |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_NZ.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.35.1"

default_platform :ios

platform :ios do

    before_all do
        update_fastlane
    end

    lane :screenshots do
        snapshot(scheme: ENV["SCHEME"], dark_mode: "true")
        snapshot(scheme: ENV["SCHEME"])
        frameit
    end

    lane :test do
        multi_scan(
            devices: ENV["DEVICES"],
            workspace: File.absolute_path('../NinetySeconds.xcworkspace'),
            scheme: ENV["SCHEME"],
	    try_count: 3
        )
    end

    lane :setup_provision do
        cert
        sigh(
            username: "iosbuild@90seconds.tv",
            app_identifier: ENV["APP_IDENTIFIER"],
            force: true
        )

    end

    lane :check_build do
        # Build app
gym(scheme: ENV["SCHEME"], skip_archive: "true", skip_package_ipa: "true", configuration: "Debug-Local-Testing", destination: "generic/platform=iOS Simulator")
    end

    lane :beta do
        # test app
        test

        # upload
        beta_upload
    end

    lane :beta_upload do
        bump_build_number

        # prepare provisioning and certificates
        setup_provision

        # build and archive
        build

        # Upload to TestFlight
        # Note: We can change this to upload_with_prompt_release_note if we don't want to use the changelog file for the release note
        upload_with_local_release_note

        # Update crashlytics DSYM file via iTunes Connect
        update_crashlytics_dsym

        version = get_build_number
    end

    lane :bump_build_number do

        # we are in detached-head state so check out the original branch to push back to
        Actions.sh("git checkout " + ENV["CI_COMMIT_REF_NAME"])
        Actions.sh("git reset --hard origin/" + ENV["CI_COMMIT_REF_NAME"])

        # Ensure that your git status is not dirty
        ensure_git_status_clean

        # Increment the build number
        increment_build_number(xcodeproj: "NinetySeconds.xcodeproj")

        # Commit the version bump
        version = get_build_number
        commit_version_bump(
            message: "Update build to " + version + ".",
            xcodeproj: "NinetySeconds.xcodeproj"
        )

        add_git_tag(tag: "testflight/6.0/" + version, force: true)

        # Push the new commit and tag back to your git remote
        push_to_git_remote(local_branch: ENV["CI_COMMIT_REF_NAME"])
    end

    lane :upload_with_local_release_note do
        # note the `..`, since fastlane runs in the _fastlane_ directory
        changelog = File.read("../Changelog.txt")

        testflight(changelog: changelog)
    end

    lane :upload_with_prompt_release_note do
        #   The user confirms their input by typing `END` and Enter
        changelog = prompt(text: "Changelog: ", multi_line_end_keyword: "END")

        testflight(changelog: changelog)
    end

    lane :update_crashlytics_dsym do
        # Download DSYM from iTunes Connect
        download_dsyms

        # Upload the downloaded DSYM file to crashlytics
        upload_symbols_to_crashlytics(api_token: ENV["CRASHLYTICS_API_TOKEN"])

        # Remove the locally downloaded DSYM file
        clean_build_artifacts
    end

end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
app_identifier ENV["APP_IDENTIFIER"] # The bundle identifier of your app
apple_id "iosbuild@90seconds.tv" # Your Apple email address

team_id "584VSD78G3" # Developer Portal Team ID

# you can even provide different app identifiers, Apple IDs and team names per lane:
# More information: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.135.2 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                      | Version | Update-Status |
| --------------------------- | ------- | ------------- |
| fastlane-plugin-test_center | 3.8.13  | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                         | Version      |
| --------------------------- | ------------ |
| did_you_mean                | 1.3.1        |
| bundler                     | 2.0.2        |
| CFPropertyList              | 3.0.1        |
| concurrent-ruby             | 1.1.5        |
| i18n                        | 0.9.5        |
| minitest                    | 5.11.3       |
| thread_safe                 | 0.3.6        |
| tzinfo                      | 1.2.5        |
| activesupport               | 4.2.11.1     |
| public_suffix               | 2.0.5        |
| addressable                 | 2.7.0        |
| atomos                      | 0.1.3        |
| babosa                      | 1.0.3        |
| claide                      | 1.0.3        |
| fuzzy_match                 | 2.0.4        |
| nap                         | 1.1.0        |
| cocoapods-core              | 1.7.5        |
| cocoapods-deintegrate       | 1.0.4        |
| cocoapods-downloader        | 1.2.2        |
| cocoapods-plugins           | 1.0.0        |
| cocoapods-search            | 1.0.0        |
| cocoapods-stats             | 1.1.0        |
| netrc                       | 0.11.0       |
| cocoapods-trunk             | 1.4.0        |
| cocoapods-try               | 1.1.0        |
| colored2                    | 3.1.2        |
| escape                      | 0.0.4        |
| fourflusher                 | 2.3.1        |
| gh_inspector                | 1.1.3        |
| molinillo                   | 0.6.6        |
| ruby-macho                  | 1.4.0        |
| nanaimo                     | 0.2.6        |
| xcodeproj                   | 1.13.0       |
| cocoapods                   | 1.7.5        |
| colored                     | 1.2          |
| colorize                    | 0.8.1        |
| highline                    | 1.7.10       |
| commander-fastlane          | 4.4.6        |
| declarative                 | 0.0.10       |
| declarative-option          | 0.1.0        |
| digest-crc                  | 0.4.1        |
| unf_ext                     | 0.0.7.6      |
| unf                         | 0.1.4        |
| domain_name                 | 0.5.20190701 |
| dotenv                      | 2.7.5        |
| emoji_regex                 | 1.0.1        |
| excon                       | 0.68.0       |
| multipart-post              | 2.0.0        |
| faraday                     | 0.17.0       |
| http-cookie                 | 1.0.3        |
| faraday-cookie_jar          | 0.0.6        |
| faraday_middleware          | 0.13.1       |
| fastimage                   | 2.1.7        |
| jwt                         | 2.1.0        |
| memoist                     | 0.16.1       |
| multi_json                  | 1.14.1       |
| os                          | 1.0.1        |
| signet                      | 0.12.0       |
| googleauth                  | 0.6.7        |
| httpclient                  | 2.8.3        |
| mime-types-data             | 3.2019.1009  |
| mime-types                  | 3.3          |
| uber                        | 0.1.0        |
| representable               | 3.0.4        |
| retriable                   | 3.1.2        |
| google-api-client           | 0.23.9       |
| google-cloud-env            | 1.3.0        |
| google-cloud-core           | 1.4.1        |
| google-cloud-storage        | 1.16.0       |
| json                        | 2.2.0        |
| mini_magick                 | 4.9.5        |
| multi_xml                   | 0.6.0        |
| plist                       | 3.5.0        |
| rubyzip                     | 1.3.0        |
| security                    | 0.1.3        |
| naturally                   | 2.2.0        |
| simctl                      | 1.6.6        |
| slack-notifier              | 2.3.2        |
| terminal-notifier           | 2.0.0        |
| unicode-display_width       | 1.6.0        |
| terminal-table              | 1.8.0        |
| tty-screen                  | 0.7.0        |
| tty-cursor                  | 0.7.0        |
| tty-spinner                 | 0.9.1        |
| word_wrap                   | 1.0.0        |
| rouge                       | 2.0.7        |
| xcpretty                    | 0.3.0        |
| xcpretty-travis-formatter   | 1.0.0        |
| xctest_list                 | 1.1.8        |
| fastlane-plugin-test_center | 3.8.13       |
</details>

*generated on:* **2019-11-15**
</details></pre>
</details>

**Comment by lyndsey-ferguson** [comment]
It does not work for me if I include the `--capture-output`, but it works fine if I do not include it. Can you run:
```
gem uninstall fastlane-plugin-test_center
```
And then run the
```
bundle exec fastlane update_plugins --verbose
```
command again?

**Comment by lyndsey-ferguson** [comment]
This ticket will close tomorrow at the end of the day (EST) due to lack of activity.

**Comment by rahulsharma2905** [comment]
@lyndsey-ferguson, i am facing the same issue on one of my machines.
output of command ```bundle exec fastlane update_plugins --verbose```
```
bundler: failed to load command: fastlane (/usr/local/bin/fastlane)
NameError: uninitialized constant #<Class:Fastlane::CLIToolsDistributor>::FastlaneCore
  /Library/Ruby/Gems/2.3.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `ensure in take_off'
  /Library/Ruby/Gems/2.3.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
  /Library/Ruby/Gems/2.3.0/gems/fastlane-2.136.0/bin/fastlane:23:in `<top (required)>'
  /usr/local/bin/fastlane:22:in `load'
  /usr/local/bin/fastlane:22:in `<top (required)>'
```

**Comment by lyndsey-ferguson** [comment]
@rahulsharma2905 this is a different issue, but the root problem may be similar.

Let's focus on `uninitialized constant`. There is a problem with your gem installation. Have you already performed the steps that I [suggested](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/issues/170#issuecomment-554098060)

**Comment by rahulsharma2905** [comment]
@lyndsey-ferguson yes i performed given steps,  and it failed on last command. Output of last command is in my last comment.

**Comment by lyndsey-ferguson** [comment]
I think that [this issue](https://github.com/fastlane/fastlane/issues/8530#issuecomment-355533052) may be the problem. Two different installations of ruby or fastlane.

I recommend using [rbenv](https://github.com/rbenv/rbenv) to manage where gems are installed.

**Comment by lyndsey-ferguson** [comment]
@rahulsharma2905 can you modify your `Pluginfile` per my instructions below, run `bundle install`, and then run your fastlane again (with the `--verbose` flag)?

Pluginfile:
```
gem 'fastlane-plugin-test_center', :git => "https://github.com/lyndsey-ferguson/fastlane-plugin-test_center.git", :branch => "issue-174-uninitialized-constant"
```

If there are still problems, please let me know and attach the console output as a text file to this issue (makes it easier for me to review).  If it works, please let me know.

**Comment by lyndsey-ferguson** [comment]
@rahulsharma2905 I do not think that my fix I mentioned above will help you for:

>  i am facing the same issue on one of my machines.
output of command bundle exec fastlane update_plugins --verbose

```
bundler: failed to load command: fastlane (/usr/local/bin/fastlane)
NameError: uninitialized constant #<Class:Fastlane::CLIToolsDistributor>::FastlaneCore
  /Library/Ruby/Gems/2.3.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `ensure in take_off'
  /Library/Ruby/Gems/2.3.0/gems/fastlane-2.136.0/fastlane/lib/fastlane/cli_tools_distributor.rb:122:in `take_off'
  /Library/Ruby/Gems/2.3.0/gems/fastlane-2.136.0/bin/fastlane:23:in `<top (required)>'
  /usr/local/bin/fastlane:22:in `load'
  /usr/local/bin/fastlane:22:in `<top (required)>'
```

This is a problem with your fastlane installation. I do _not_ recommend using your System Ruby Gems directories to install non-Apple gems. Problems generally result.
(`/Library/Ruby/Gems/2.3.0`).

I highly recommend using [`rbenv`](https://github.com/rbenv/rbenv)

**Comment by rahulsharma2905** [comment]
@lyndsey-ferguson , i actually use --path in bundle exec command, which lets me install gems in a local directory separate from System Ruby Gems.
when i tried command ```bundle install --path vendor/bundle```, it installs all gems in vendor/bundle. and the output of this command is throwing the same error as mentioned in this issue.
```
Error loading plugin 'fastlane-plugin-test_center': uninitialized constant TestCenter::Helper::MultiScanManager::TestBatchWorker
```
P.S. i tried changing my Plugin file to ```gem 'fastlane-plugin-test_center', :git => "https://github.com/lyndsey-ferguson/fastlane-plugin-test_center.git", :branch => "issue-174-uninitialized-constant"```, but still facing the same error.

**Comment by lyndsey-ferguson** [comment]
I've made some changes. Can you try updating your `Gemfile.lock` to use this git SHA: `c0ba3dbaa19e234f3f59f3ea744a815acea683f1`, bundle install, and re-run your lane?

**Comment by rahulsharma2905** [comment]
@lyndsey-ferguson , 👍 it worked. The error is gone, and tests are running fine.

**Comment by lyndsey-ferguson** [comment]
# 🎉 
# 🎉 
# 🎉

**Comment by lyndsey-ferguson** [comment]
Fixed in [v3.8.15](https://github.com/lyndsey-ferguson/fastlane-plugin-test_center/releases/tag/v3.8.15)
